ultra/ultra/baselines/rllib/agent.py

```diff
@@ -0,0 +1,72 @@
+# MIT License
+#
+# Copyright (C) 2021. Huawei Technologies Co., Ltd. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+import ray.rllib.agents.ppo as ppo
+import ray.rllib.agents.sac as sac
+import ray.rllib.agents.ddpg as ddpg
+
+
+class RllibAgent:
+    def __init__(self, agent_type, env, config, logger_creator):
+        assert agent_type in ["td3", "ddpg", "dqn", "ppo", "sac"]
+        self.agent_type = agent_type
+
+        if self.agent_type == "ppo":
+            self.trainer = ppo.PPOTrainer(
+                env=env,
+                config=config,
+                logger_creator=logger_creator,
+            )
+        elif self.agent_type == "ddpg":
+            self.trainer = ddpg.DDPGTrainer(
+                env=env,
+                config=config,
+                logger_creator=logger_creator,
+            )
+        elif self.agent_type == "td3":
+            self.trainer = td3.TD3Trainer(
+                env=env,
+                config=config,
+                logger_creator=logger_creator,
+            )
+        # elif self.agent_type=='dqn':
+        #     trainer = dqn.DQNTrainer(
+        #         env=RLlibUltraEnv,
+        #         config=tune_config,
+        #         logger_creator=log_creator(log_dir),
+        #     )
+
```

Comment:
Should we raise an error if `agent_type` is in `["td3", ..., "sac"]` but is not one of our currently supported trainer types (PPO, DDPG, TD3)? E.g. Should we raise an error if someone wants `agent_type=sac` or `agent_type=dqn`?

Reply:
Hmm, yes that would be needed. I do not know currently if @adreena has managed to get the other policy to work though